### PR TITLE
French translation

### PR DESCRIPTION
**Pull request by abulte**
A French translation for `flask_security`, sponsored by https://github.com/etalab.

It would be great if this could be included in the (much anticipated) next release!

Have a good day.

### CODE REVIEW

File: flask_security/translations/fr_FR/LC_MESSAGES/flask_security.po [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+msgid "Reset password"
+msgstr "Réinitialiser le mot de passe"
+
+#: flask_security/templates/security/send_confirmation.html:3
+msgid "Resend confirmation instructions"
+msgstr "Renvoyer les instructions de confirmation"
+
+#: flask_security/templates/security/email/change_notice.html:1
+msgid "Your password has been changed."
+msgstr "Votre mot de passe a été changé"
```
Comment (vinyll) [inline flask_security/translations/fr_FR/LC_MESSAGES/flask_security.po]: missing a "."

File: flask_security/translations/fr_FR/LC_MESSAGES/flask_security.po [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+#: flask_security/templates/security/email/confirmation_instructions.html:3
+#: flask_security/templates/security/email/welcome.html:6
+msgid "Confirm my account"
+msgstr "Confirmer mon compte"
+
+#: flask_security/templates/security/email/login_instructions.html:1
+#: flask_security/templates/security/email/welcome.html:1
+#, python-format
+msgid "Welcome %(email)s!"
+msgstr "Bienvenue %(email)s !"
```
Comment (jirikuncar) [inline flask_security/translations/fr_FR/LC_MESSAGES/flask_security.po]: an extra space before `!`

File: flask_security/translations/fr_FR/LC_MESSAGES/flask_security.po [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+
+#: flask_security/templates/security/email/login_instructions.html:3
+msgid "You can log into your account through the link below:"
+msgstr "Vous pouvez vous connecter via le lien ci-dessous :"
```
Comment (jirikuncar) [inline flask_security/translations/fr_FR/LC_MESSAGES/flask_security.po]: space before `:`

File: flask_security/translations/fr_FR/LC_MESSAGES/flask_security.po [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+msgid "Login now"
+msgstr "Se connecter maintenant"
+
+#: flask_security/templates/security/email/reset_instructions.html:1
+msgid "Click here to reset your password"
+msgstr "Cliquez pour réinitialiser votre mot de passe"
+
+#: flask_security/templates/security/email/welcome.html:4
+msgid "You can confirm your email through the link below:"
+msgstr "Vous pouvez confirmer votre votre adresse email via le lien ci-dessous :"
```
Comment (jirikuncar) [inline flask_security/translations/fr_FR/LC_MESSAGES/flask_security.po]: see https://github.com/mattupstate/flask-security/pull/657/files#r121054788

File: flask_security/translations/fr_FR/LC_MESSAGES/flask_security.po (hunk 1 above)
Comment (abulte) [inline flask_security/translations/fr_FR/LC_MESSAGES/flask_security.po]: 👍

**Comment by abulte** [comment]
@jirikuncar about the spaces, it's the way we're supposed to write in French, see https://www.thoughtco.com/how-to-use-french-punctuation-4086509

> In French, a space is required both before and after all two- (or more) part punctuation marks and symbols, including : ; « » ! ? % $ #

I will add the missing dot.

**Comment by vinyll** [comment]
@jirikuncar In English, like most – not to say all – languages, there's no non-breakable space before the list of punctuations that @abulte mentioned.
French must have them and it's often a way to recognize French typing in English, they add spaces everywhere ^^

**Comment by jirikuncar** [comment]
Interesting, I have made some research about it and it looks you are right with space, but it should be a "une fine" (HTML = &thinsp; / Unicode = U+2009) / "une espace insecable" (HTML = &nbsp; / Unicode = U+00A0).

The Canadians have a slightly different point of view http://www.btb.termiumplus.gc.ca/tcdnstyl-chap?lang=eng&lettr=chapsect17&info0=17

**Comment by abulte** [comment]
@jirikuncar you're totally right, it's supposed to be a "Espace fine insécable". I usually avoid them by fear of unexpected side effects though... Do you want me to try anyway?

**Comment by jirikuncar** [comment]
@abulte we should try so there are no unexpected results in the UI like hanging `:` on new line.

File: flask_security/translations/fr_FR/LC_MESSAGES/flask_security.po (hunk 2 above)
Comment (davidbgk) [inline flask_security/translations/fr_FR/LC_MESSAGES/flask_security.po]: This is intended, that is the [way to do in French](https://www.thoughtco.com/how-to-use-french-punctuation-4086509). Not sure if it's a non-breaking space though. @abulte can you check/fix if that's not the case?

**Comment by abulte** [comment]
@jirikuncar ok I'll do that.

File: flask_security/translations/fr_FR/LC_MESSAGES/flask_security.po (hunk 2 above)
Comment (odtvince) [inline flask_security/translations/fr_FR/LC_MESSAGES/flask_security.po]: That extra (non breaking) space is indeed required in French typography

File: flask_security/translations/fr_FR/LC_MESSAGES/flask_security.po (hunk 3 above)
Comment (odtvince) [inline flask_security/translations/fr_FR/LC_MESSAGES/flask_security.po]: That extra (non breaking) space is indeed required in French typography

**Comment by abulte** [comment]
Done! `&thinsp;` everywhere. While testing, I also encountered and fixed a bug with non latin 1 characters on the forgot password field description (was failing for Russian and French at least).

**Comment by abulte** [comment]
Thanks @jirikuncar!

**Pull request merged**